source/dub/commandline.d

```diff
@@ -77,6 +78,300 @@ CommandGroup[] getCommands()
 	];
 }
 
+/** Extract the command name from the argument list
+*/
+struct CommandNameArgument
+{
+	private string m_value;
+	private string[] m_args;
+
+	/** Separates the command name from the arguments.
+	*/
+	this(string[] args)
+	{
+		if (args.length >= 1 && !args[0].startsWith("-")) {
+			m_value = args[0];
+			m_args = args[1 .. $];
+		} else {
+			m_args = args;
+		}
+	}
+
+	/** Returns the command name.
+	*/
+	string value()
+	{
+		return m_value;
+	}
+
+	/** Returns the list of unprocessed arguments and calls `dropAllArgs`.
+	*/
+	string[] extractAllRemainingArgs()
+	{
+		return m_args;
+	}
+}
+
+/// It returns an empty string on when there are no args
+unittest {
+	assert(CommandNameArgument([]).value == "");
+}
+
+/// It returns the first argument when it does not start with `-`
+unittest {
+	assert(CommandNameArgument(["test"]).value == "test");
+}
+
+/// There is nothing to extract when the arguments only contain the `test` cmd
+unittest {
+	assert(CommandNameArgument(["test"]).extractAllRemainingArgs == []);
+}
+
+/// It extracts two arguments when they are not a command
+unittest {
+	assert(CommandNameArgument(["-a", "-b"]).extractAllRemainingArgs == ["-a", "-b"]);
+}
+
+/// It returns the an empty string when it starts with `-`
+unittest {
+	assert(CommandNameArgument(["-test"]).value == "");
+}
+
+/** Handles the Command Line options and commands.
+*/
+struct CommandLineHandler
+{
+	/// The list of commands that can be handled
+	CommandGroup[] commands;
+
+	/// General options parser
+	CommonOptions options;
+
+	/** Create the list of all supported commands
+
+		Returns:
+			Returns the list of the supported command names
+	*/
+	string[] commandNames()
+	{
+		return commands.map!(g => g.commands.map!(c => c.name).array).join.array;
+	}
+
+	/** Parses the general options and sets up the log level
+		and the root_path
+	*/
+	void prepareOptions(CommandArgs args) {
+		LogLevel loglevel = LogLevel.info;
+
+		options.prepare(args);
+
+		if (options.vverbose) loglevel = LogLevel.debug_;
+		else if (options.verbose) loglevel = LogLevel.diagnostic;
```

Comment:
dupilcate argument to `if` an `else if`

Reply:
for which values? here is `vverbose` and `verbose`

Reply:
Ah.